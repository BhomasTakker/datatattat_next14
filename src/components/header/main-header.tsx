import { ClientHeader } from "./client-header";
import { NavigationHeader } from "./navigation-header";
import styles from "./main-header.module.scss";

export interface DynamicHeader {
	route?: string[] | undefined;
}
export const MainHeader = ({ route }: DynamicHeader) => {
	return (
		<header className={styles.root}>
			<NavigationHeader />
			<ClientHeader route={route} />
		</header>
	);
};
