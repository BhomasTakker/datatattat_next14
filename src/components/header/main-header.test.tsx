import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { MainHeader } from "./main-header";

jest.mock("./navigation-header", () => {
	return {
		NavigationHeader: () => {
			return <h2>Navigation Header</h2>;
		},
	};
});

jest.mock("./client-header", () => {
	return {
		ClientHeader: () => {
			return <h2>Client Header</h2>;
		},
	};
});

describe("MainHeader", () => {
	it("renders a header component", () => {
		render(<MainHeader />);
		const header = screen.getByRole("banner");
		expect(header).toBeInTheDocument();
	});

	it("renders the NavigationHeader", () => {
		render(<MainHeader />);
		const mainHeader = screen.queryByText("Navigation Header");
		expect(mainHeader).toBeInTheDocument();
	});

	it("renders the ClientHeader", () => {
		render(<MainHeader />);
		const clientHeader = screen.queryByText("Client Header");
		expect(clientHeader).toBeInTheDocument();
	});
});

describe("MainHeader snaps", () => {
	it("renders MainHeader unchanged", () => {
		const { container } = render(<MainHeader />);
		expect(container).toMatchSnapshot();
	});
});
