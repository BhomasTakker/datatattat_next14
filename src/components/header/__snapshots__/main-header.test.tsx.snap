// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainHeader snaps renders MainHeader unchanged 1`] = `
<div>
  <header
    class="root"
  >
    <div
      class="container"
    >
      <h2>
        Navigation Header
      </h2>
    </div>
    <div
      class="divider"
    />
    <div
      class="container"
    >
      <h2>
        Client Header
      </h2>
    </div>
    <div
      class="divider"
    />
  </header>
</div>
`;
